### ISSUE TITLE
train the densecl model on my custom dataset

### ISSUE THREAD

**Issue by alaa-shubbak** [open-mmlab/mmpretrain#1541]
### Describe the feature

I did the training on densecl model on my custom dataset . 
i make the max_epoch first to be 300 as my dataset is very small ( only 170 images for training) , the increase the max_epoch to become 800 to see if the loss will decrease. 

i noticed that both loss and loss_dense kept to increase as the following image . 

![dense_train1](https://user-images.githubusercontent.com/10245810/236251769-0f0bd987-817d-46a5-a14c-04e40c55f08a.jpg)
![dense_train2](https://user-images.githubusercontent.com/10245810/236251784-0776c392-8879-4970-851e-7eb3d5b21331.jpg)

is that true ? is there any explanation for that? 
i used another self-supervised model (simclr) with my same custom dataset , but the loss there are decreasing 

any suggestion please ?  

### Will you implement it?

- [ ] I would like to implement this feature and create a PR!

**Comment by fangyixiao18** [comment]
For moco and densecl (the queue mechanism also inherits from moco), in the beginning, the queue saves samples and the samples in the queue are not stable. 
The loss computation is based on samples saved in the queue and the current batch, it is confusing for model to do discrimination with random vectors.
https://github.com/open-mmlab/mmpretrain/blob/main/mmpretrain/models/selfsup/densecl.py#L74

You could also check our log of moco and densecl, the loss is also increasing at the beginning.
https://download.openmmlab.com/mmselfsup/1.x/densecl/densecl_resnet50_8xb32-coslr-200e_in1k/densecl_resnet50_8xb32-coslr-200e_in1k_20220825-3078723b.json

**Comment by alaa-shubbak** [comment]
thanks for reply. 
I trained the model again with much huge dataset. 
i got this as plot from the loss of densecl model. 

<img width="600" alt="image" src="https://user-images.githubusercontent.com/10245810/236552429-79815c96-580b-4898-bb4c-647a17009f1a.png">

and this for the loss single of densecl 
<img width="619" alt="image" src="https://user-images.githubusercontent.com/10245810/236552644-7bf1a0b6-0ab1-45c2-b474-59c5c7496a79.png">

in such situation shall i take the model results at an epoch around 200 as the optimal solution and use it later on my downstream task?

**Comment by fangyixiao18** [comment]
the model at 200 epochs may not be the optimal solution, as your dataset is very small, here are some advice you may try for your downstream tasks:
1. try the model at 200 epochs
2. try the model at 800 epochs
3. try the model we provide, which is pre-trained on ImageNet
4. reduce the `queue_len` according to your datasets and re-train the densecl and use the model

**Comment by alaa-shubbak** [comment]
> 4\. queue_len

where can i change this ? and what does it mean?
